src/main.zig

```diff
@@ -2846,21 +2892,17 @@ fn fmtPathFile(
     if (stat.kind == .Directory)
         return error.IsDir;
 
-    const source_code = source_file.readToEndAllocOptions(
+    const source_code = readSourceFileToEndAlloc(
         fmt.gpa,
-        max_src_size,
+        &source_file,
         std.math.cast(usize, stat.size) catch return error.FileTooBig,
-        @alignOf(u8),
-        null,
-    ) catch |err| switch (err) {
-        error.ConnectionResetByPeer => unreachable,
-        error.ConnectionTimedOut => unreachable,
-        error.NotOpenForReading => unreachable,
-        else => |e| return e,
+    ) catch |err| {
+        fatal("unable to read file {s}: {s}", .{ file_path, err });
```

Comment:
Do not use `fatal` here, we don't want to stop formatting all the files in a directory just because one of them is using an unsupported encoding.

Reply:
Thank you for the review ! I did not see the error was logged anyway in `fmtPathDir`. The error is now just returned.